UnityProject/Assets/Scripts/Player/JoinedViewer.cs

```diff
@@ -308,45 +308,85 @@ public void ClientFinishLoading()
 			}
 		}
 
 		/// <summary>
 		/// Waits for the client to be an observer of the player before continuing
 		/// </summary>
-		private IEnumerator WaitForLoggedOffObserver(Mind loggedOffPlayer)
+		private async UniTask WaitForLoggedOffObserver(Mind loggedOffPlayer)
 		{
 			TargetLocalPlayerRejoinUI(connectionToClient);
 			// TODO: When we have scene network culling we will need to allow observers
 			// for the whole specific scene and the body before doing the logic below:
-			var netIdentity = loggedOffPlayer.GetComponent<NetworkIdentity>();
-			if (netIdentity == null)
+			var identity = loggedOffPlayer.GetComponent<NetworkIdentity>();
+			if (identity == null)
 			{
 				GUI_PreRoundWindow.Instance?.OnClientLoadUpdateStatus?.Invoke("An error occurred. Press F5 to check for what error had occured.".Color(Color.red));
 				Loggy.Error($"No {nameof(NetworkIdentity)} component on {loggedOffPlayer}! " +
 				                "Cannot rejoin that player. Was original player object improperly created? " +
 				                "Did we get runtime error while creating it?", Category.Connections);
 				// TODO: if this issue persists, should probably send the poor player a message about failing to rejoin.
 				ClearCache();
-				yield break;
+				return;
 			}
 
-			while (!netIdentity.observers.ContainsKey(connectionToClient.connectionId))
+			var antiFreezeCheckCount = 0;
+			while (identity.observers.ContainsKey(connectionToClient.connectionId) == false)
 			{
-				yield return WaitFor.EndOfFrame;
+				antiFreezeCheckCount++;
+				await UniTask.WaitForEndOfFrame();
 				if (connectionToClient == null)
 				{
 					//disconnected while we were waiting
 					ClearCache();
-					yield break;
+					break;
+				}
+				if (antiFreezeCheckCount > 5500)
+				{
+					GUI_PreRoundWindow.Instance?.OnClientLoadUpdateStatus?.Invoke("A problem occurred while attempting to check for a valid connection ID." +
+						"No valid connection found after 55000 frames. Press F5 to check for if an error had occured.".Color(Color.red));
+					Loggy.Error($"ID {connectionToClient.connectionId} not found in observers dictionary!" +
+					            "Cannot rejoin that player. Was original player object improperly created? " +
+					            "Did we get runtime error while creating it?");
+					//FIXME: This is a temporary banadge for a game breaking issue.
+					//(Max): I can't figure out why the observers dictionary isn't getting updated accordingly, what is responsible for it.
+					//This way of checking possesion IDs directly should at least stop players from getting stuck on round-rejoins,
+					//but it isn't encourged to be the main way of handling this.
+					AttemptFallback(loggedOffPlayer, connectionToClient);
+					ClearCache();
+					return;
 				}
 			}
 
+			SuccesfullyRejoin();
+		}
+
+		[Command]
+		private void AttemptFallback(Mind loggedOffPlayer, NetworkConnectionToClient conn = null)
+		{
+			if (conn == null) return; //weaver requirement
```

Comment:
Got to validate in case a hacked client calls this early.

Reply:
How would we validate this?

Reply:
IsValidPlayerAndWaitingOnLoad

Reply:
updated it to use that